assets/js/WorkflowEditor/AddData/Modal.js

```diff
@@ -8,22 +8,22 @@ import { connect } from 'react-redux'
 import Modules from './Modules'
 import Search from './Search'
 import { Trans, t } from '@lingui/macro'
-import { I18n } from '@lingui/react'
-
-export const Modal = React.memo(function Modal ({ modules, tabSlug, close, addModule }) {
+import { withI18n } from '@lingui/react'
+export const Modal = React.memo(function Modal ({ i18n, modules, tabSlug, close, addModule }) {
   const onSelectModule = React.useCallback(moduleIdName => addModule(tabSlug, moduleIdName))
   const [search, setSearch] = React.useState('')
 
   return (
     <section className='add-data-modal'>
       <header>
         <div className='title'>
-          <h5><Trans id='workflow.choosedata'>CHOOSE A DATA SOURCE</Trans></h5>
-          <I18n>
-            {({ i18n }) => (
-              <button type='button' className='close' aria-label='Close' title={i18n._(t('workflow.attribute.close')`Close`)} onClick={close}>×</button>
-            )}
-          </I18n>
+          <h5>
+            <Trans id='js.WorkflowEditor.AddData.Modal.header.title' description='This should be all-caps for styling reasons'>
+                CHOOSE A DATA SOURCE
+            </Trans>
+          </h5>
+
+          <button type='button' className='close' aria-label='Close' title={i18n._(t('js.WorkflowEditor.AddData.Modal.closeButton.hoverText')`Close`)} onClick={close}>×</button>
```

Comment:
Regarding this and the other `.hoverText`s:
When somebody sees `js.WorkflowEditor.AddData.Modal.closeButton.title` in the translation file, they may misunderstand the meaning of `title`. That's why I avoided it and used something that describes the function of the text better. If you think we should use `title`, an i18n comment should definitely be added.

Reply:
Good point about `.title` missing key information. My vote is for `.title` + a comment, as you suggest.

I don't like the idea of an "almost-HTML" pattern, because it'll be hard for developers to adhere to it. It leads to more ambiguities, such as, "what do we call the `.title` in an `<abbr>` element? Then logistically, where do we document all our quasi-HTML attributes?

I think it's reasonable to teach translators that an HTML `title` is text that appears on hover. 
HTML is unintuitive. But I doubt our message-ID system can be a _significantly_ better.

@giorgio93p I'm willing to go with `.hoverText` if you feel more strongly than I do about this :).